react-auth-client/src/components/home.jsx

```diff
@@ -12,116 +13,159 @@ class Home extends React.Component {
-    let nonce = api.generateRandomValue();
+    let state = authClient.generateRandomValue();
+    let nonce = authClient.generateRandomValue();
+    // Store state and nonce parameters into the session, so we can retrieve them after
+    // user will be redirected back with access token or code (since react state is cleared in this case)
     sessionStorage.setItem("state", state);
+    sessionStorage.setItem("nonce", nonce);
 
-    api.authorize(authDetails.environmentId,
-        authDetails.responseType, authDetails.clientId,
-        authDetails.redirectUri, authDetails.scope,
-        state, nonce,
-        authDetails.prompt, authDetails.maxAge);
-
+    authClient.authorize(state, nonce);
   }
 
   handleSignOff() {
-    if (sessionStorage.getItem("id_token")) {
-      api.signOff(this.props.authDetails.environmentId,
-          this.props.authDetails.logoutRedirectUri,
-          sessionStorage.getItem("id_token"), sessionStorage.getItem("state"));
+    if (this.state.id_token) {
+      authClient.signOff(this.state.id_token, sessionStorage.getItem("state"));
     }
-
     this.clearSession();
   }
 
   clearSession() {
-    sessionStorage.removeItem("access_token");
-    sessionStorage.removeItem("id_token");
-    sessionStorage.removeItem("code");
-    sessionStorage.removeItem("expires_in");
-    sessionStorage.removeItem("scope");
-    sessionStorage.removeItem("state");
+    this.setState({
+      access_token: null,
+      id_token: null,
+      errorMessage: ''
+    });
   }
 
   componentDidMount() {
-    const {authDetails} = this.props;
-
-    const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
-
-    if (!uuidRegex.test(authDetails.environmentId)) {
-      this.setState({
-        errorMessage: `Invalid environmentId parameter ${authDetails.environmentId} : it should be a valid UUID.  Please check it in your config.js parameters file.`,
-      });
-    }
-
-    let hashes = api.parseHash();
-
-    if (hashes && hashes.error && hashes.error_description) {
+    const hashes = authClient.parseHash();
+    if (hashes.error && hashes.error_description) {
       this.setState({
         errorMessage: hashes.error + ': ' + hashes.error_description,
       });
       return;
     }
 
-    let stateMatch = window.location.href.match('[?#&]state=([^&]*)');
+    const stateMatch = window.location.href.match('[?#&]state=([^&]*)');
     if (stateMatch && !stateMatch[1] &&
         !_.isEqual(stateMatch[1], sessionStorage.getItem("state"))) {
       this.setState({
-        errorMessage: "State parameter mismatch"
+        errorMessage: "State parameter mismatch. "
       });
+      this.clearSession();
       return;
     }
 
-    let codeMatch = window.location.href.match('[?#&]code=([^&]*)');
+    if (hashes.access_token) {
+      this.setState({
+        access_token: hashes.access_token
+      })
+      this.handleUserInfo(hashes.access_token);
+    } else {
```

Comment:
feels weird that this is an else vs if (codeMatch)

Reply:
Refactored to
```
    const codeMatch = window.location.href.match('[?#&]code=([^&]*)');
    // Implicit flow: access token is present in URL
    if (hashes.access_token) {
       .....
    }
    // Authorization code flow: access code is present in URL
    else if (codeMatch && codeMatch[1]) {
```

Reply:
Nice!